src/lib.rs

```diff
@@ -169,28 +193,41 @@ pub fn get_public_key(
     match transport.exchange(&command) {
         Ok(response) => {
             log::info!(
-                "APDU out: {}\nAPDU ret code: {:x}, {:?}",
+                "APDU out: {}\nAPDU ret code: {:x}",
                 hex::encode(response.apdu_data()),
                 response.retcode(),
-                response.error_code()
             );
             // Ok means we successfully exchanged with the Ledger
             // but doesn't mean our request succeeded
             // we need to check it based on `response.retcode`
             if response.retcode() == RETURN_CODE_OK {
                 return Ok(ed25519_dalek::PublicKey::from_bytes(&response.data()).unwrap());
             } else {
-                let error_string = response
-                    .error_code()
-                    .map(|error_code| error_code.to_string())
-                    .unwrap_or_else(|retcode| format!("Unknown Ledger APDU retcode: {}", retcode));
+                let retcode = response.retcode();
+
+                let error_string = format!("Ledger APDU retcode: 0x{:X}", retcode);
                 return Err(NEARLedgerError::APDUExchangeError(error_string));
             }
         }
         Err(err) => return Err(NEARLedgerError::LedgerHIDError(err)),
     };
 }
 
+fn get_transport() -> Result<TransportNativeHID, NEARLedgerError> {
+    // instantiate the connection to Ledger
+    // will return an error if Ledger is not connected
+    let hidapi = match HidApi::new() {
+        Ok(hidapi) => hidapi,
+        Err(_err) => return Err(NEARLedgerError::DeviceNotFound),
+    };
+    match TransportNativeHID::new(&hidapi) {
+        Ok(transport) => Ok(transport),
+        // TODO: refactor this
+        // https://github.com/Zondax/ledger-rs/issues/65
+        Err(_err) => Err(NEARLedgerError::DeviceNotFound),
```

Comment:
Since the issue is resolved, can we address this TODO?

Reply:
addressed in [commit](https://github.com/khorolets/near-ledger-rs/pull/4/commits/25b16bd49998b08d85aba6ce177ce453dc2c4948)